scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2116 @@
+    project_settings = json.loads(project_json_content)
+    # Read the android_project.json next. If it does not exist, check if this is a legacy project where the android
+    # settings are in the root project.json
+    android_project_json_file = project_path / 'Platform' / 'Android' / 'android_project.json'
+    if android_project_json_file.is_file():
+        android_project_json_content = android_project_json_file.read_text(encoding=DEFAULT_READ_ENCODING,
+                                                                           errors=ENCODING_ERROR_HANDLINGS)
+        android_json = json.loads(android_project_json_content)
+        android_settings = android_json.get('android_settings')
+        if android_settings is None:
+            raise AndroidToolError(f"Missing android settings in file {android_project_json_file}")
+        android_settings = android_json['android_settings']
+    else:
+        android_settings = project_settings.get('android_settings')
+        if android_settings is None:
+            raise AndroidToolError(f"Missing android settings file {android_project_json_file} and cannot located legacy "
+                                   f"'android_settings' from {project_json_path}")
+
+    return project_settings, android_settings
+
+
+class AndroidSigningConfig(object):
+    """
+    Class to manage android signing configs section in a gradle build script
+    """
+    def __init__(self, store_file:Path, store_password:str, key_alias:str, key_password:str):
+        """
+        Initialize this android signing config object
+
+        :param store_file:      The path to the key store file used for the signing configuration
+        :param store_password:  The password to the key store file
+        :param key_alias:       The alias of the private signing key in the key store
+        :param key_password:    The password to the private signing key in the key store referenced by the key alias
+        """
+
+        self._store_file = Path(store_file)
+        self._store_password = store_password
+        self._key_alias = key_alias
+        self._key_password = key_password
+
+    def to_template_string(self, tabs) -> str:
+        """
+        Generate a string that represents a signing config section that can be inserted into a gradle script.
+
+        :param tabs:    The number of leading tabs for each line to insert for better readability of the gradle script
+        :return:    The signing config string section to insert
+        """
+        tab_prefix = ' '* 4 * tabs  # 4 spaces per tab
+        return f"{tab_prefix}storeFile file('{self._store_file.as_posix()}')\n" \
+               f"{tab_prefix}storePassword '{self._store_password}'\n" \
+               f"{tab_prefix}keyPassword '{self._key_password}'\n" \
+               f"{tab_prefix}keyAlias '{self._key_alias}'\n"
+
+
+JAVA_VERSION_REGEX = re.compile(r'.*(\w)\s(version)\s*\"?([\d\_\.]+)', re.MULTILINE)
```

Comment:
python regex supports [named groups](https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups) so you can capture the regex into a specific name
```suggestion
JAVA_VERSION_REGEX = re.compile(r'.*(\w)\s(version)\s*\"?(?P<version>[\d\_\.]+)', re.MULTILINE)
```

Reply:
Good suggestion